polaris-core/src/main/java/org/apache/polaris/core/entity/PolarisPrincipalSecrets.java

```diff
@@ -108,11 +155,29 @@ public String getPrincipalClientId() {
     return principalClientId;
   }
 
+  public boolean matchesSecret(String potentialSecret) {
+    String potentialSecretHash = hashSecret(potentialSecret);
+    return potentialSecretHash.equals(this.mainSecretHash)
+        || potentialSecretHash.equals(this.secondarySecretHash);
```

Comment:
Do we want to support rolling upgrades (when requests may submit the old plain text secrets for validation)?

Reply:
+1, I am not clear on this. I put a note in the description, but I basically copied what I saw as the existing behavior. I am open to keeping/changing this.

Reply:
Sounds like an easy enough thing to add for now to prevent a backwards incompatibility, but maybe we should just only do so if the "hash" is not set? Then the moment hash is set only accept a hashed comparison?

Reply:
Well, clients will take time to update their configuration. How about we allow plain test password even when a hash is set, but until the next password rotation? Meaning: remove plain test passwords during rotation if hash is set.

Reply:
(This is just an idea from my side. I do not mean to say that supporting plain text passwords and rolling upgrades are required from my POV :) )

Reply:
The way it works right now is this:
1. As soon as you `rotate-credentials`, the plaintext secret is wiped from the metastore. Atomically with this, the hash of that secret is added as the secondary hash
2. You can still authenticate with the old credential (it checks against that hash) until you `rotate-credentials` again, at which point the secondary secret is gone. The client still sends the full secret, but it's not persisted.

I think this is the correct behavior ***iff** we want to have secrets be valid after one `rotate-credentials` but not after two.

Reply:
@eric-maynard : Apologies, I misinterpreted the code. You're right in how it works. I think the PR is good to merge.